Runtime/src/main/java/com/rakuten/tech/mobile/perf/runtime/internal/Util.java

```diff
@@ -1,18 +1,28 @@
 package com.rakuten.tech.mobile.perf.runtime.internal;
 
 import android.content.Context;
+import android.content.pm.PackageInfo;
 import android.content.pm.PackageManager;
+import android.content.pm.Signature;
 import android.os.Bundle;
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 
+import java.io.ByteArrayInputStream;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+
+import javax.security.auth.x500.X500Principal;
+
 /**
  * Shared functions for Relay SDKs
  */
-class Util {
+public class Util {
```

Comment:
this class should **not** be public. If I see the changes correctly it's only used by `RuntimeContentProvider` which lives in the same package - so default visibility should be enough. Same applies to the `isAppDebuggable` function.

- [ ] reduce visibility of class
- [ ] reduce visibility of function

Note that it is important to not expose classes publicly unnecessarily, because that adds to the surface area of the library. Every public method we put out there we need to keep around and maintain until the next major version, because we follow semantic versioning.

Reply:
As discussed offline:
* `public` is necessary for shadowing
* use [`VisibleForTesting`](https://developer.android.com/reference/android/support/annotation/VisibleForTesting.html) and [`RestrictTo`](https://developer.android.com/reference/android/support/annotation/RestrictTo.html) annotations

- [ ] Annotate class as internal